scripts/install-worker.sh

```diff
@@ -240,88 +243,87 @@ sudo mkdir -p /var/lib/kubernetes
-    sudo wget $S3_URL_BASE/$binary
-    sudo wget $S3_URL_BASE/$binary.sha256
+    sudo wget "${S3_URL_BASE}/${binary}"
+    sudo wget "${S3_URL_BASE}/${binary}.sha256"
   fi
-  sudo sha256sum -c $binary.sha256
-  sudo chmod +x $binary
-  sudo mv $binary /usr/bin/
+  sudo sha256sum -c "${binary}.sha256"
+  sudo chmod +x "${binary}"
+  sudo mv "${binary}" /usr/bin/
 done
 
 # Verify that the aws-iam-authenticator is at last v0.5.9 or greater. Otherwise, nodes will be
 # unable to join clusters due to upgrading to client.authentication.k8s.io/v1beta1
 iam_auth_version=$(sudo /usr/bin/aws-iam-authenticator version | jq -r .Version)
-if vercmp "$iam_auth_version" lt "v0.5.9"; then
+if vercmp "${iam_auth_version}" lt "v0.5.9"; then
   # To resolve this issue, you need to update the aws-iam-authenticator binary. Using binaries distributed by EKS
   # with kubernetes_build_date 2022-10-31 or later include v0.5.10 or greater.
-  echo "❌ The aws-iam-authenticator should be on version v0.5.9 or later. Found $iam_auth_version"
+  echo "❌ The aws-iam-authenticator should be on version v0.5.9 or later. Found ${iam_auth_version}"
   exit 1
 fi
 
 # Since CNI 0.7.0, all releases are done in the plugins repo.
 CNI_PLUGIN_FILENAME="cni-plugins-linux-${ARCH}-${CNI_PLUGIN_VERSION}"
 
-if [ "$PULL_CNI_FROM_GITHUB" = "true" ]; then
+if [ "${PULL_CNI_FROM_GITHUB}" = "true" ]; then
   echo "Downloading CNI plugins from Github"
   wget "https://github.com/containernetworking/plugins/releases/download/${CNI_PLUGIN_VERSION}/${CNI_PLUGIN_FILENAME}.tgz"
   wget "https://github.com/containernetworking/plugins/releases/download/${CNI_PLUGIN_VERSION}/${CNI_PLUGIN_FILENAME}.tgz.sha512"
-  sudo sha512sum -c "${CNI_PLUGIN_FILENAME}.tgz.sha512"
+  sha512sum -c "${CNI_PLUGIN_FILENAME}.tgz.sha512"
   rm "${CNI_PLUGIN_FILENAME}.tgz.sha512"
 else
-  if [[ -n "$AWS_ACCESS_KEY_ID" ]]; then
+  if [[ -n "${AWS_ACCESS_KEY_ID}" ]]; then
     echo "AWS cli present - using it to copy binaries from s3."
-    aws s3 cp --region $BINARY_BUCKET_REGION $S3_PATH/${CNI_PLUGIN_FILENAME}.tgz .
-    aws s3 cp --region $BINARY_BUCKET_REGION $S3_PATH/${CNI_PLUGIN_FILENAME}.tgz.sha256 .
+    aws s3 cp --region "${BINARY_BUCKET_REGION}" "${S3_PATH}/${CNI_PLUGIN_FILENAME}.tgz" .
+    aws s3 cp --region "${BINARY_BUCKET_REGION}" "${S3_PATH}/${CNI_PLUGIN_FILENAME}.tgz.sha256" .
   else
     echo "AWS cli missing - using wget to fetch cni binaries from s3. Note: This won't work for private bucket."
-    sudo wget "$S3_URL_BASE/${CNI_PLUGIN_FILENAME}.tgz"
-    sudo wget "$S3_URL_BASE/${CNI_PLUGIN_FILENAME}.tgz.sha256"
+    wget "$S3_URL_BASE/${CNI_PLUGIN_FILENAME}.tgz"
+    wget "$S3_URL_BASE/${CNI_PLUGIN_FILENAME}.tgz.sha256"
   fi
-  sudo sha256sum -c "${CNI_PLUGIN_FILENAME}.tgz.sha256"
+  sha256sum -c "${CNI_PLUGIN_FILENAME}.tgz.sha256"
+  rm "${CNI_PLUGIN_FILENAME}.tgz.sha256"
 fi
 sudo tar -xvf "${CNI_PLUGIN_FILENAME}.tgz" -C /opt/cni/bin
 rm "${CNI_PLUGIN_FILENAME}.tgz"
 
-sudo rm ./*.sha256
```

Comment:
moved to Line-307.

Reply:
This line is intended to remove *all* checksum files, including the ones downloaded in this block: https://github.com/awslabs/amazon-eks-ami/blob/master/scripts/install-worker.sh#L259-L272

So, moving this to 307 will leave behind those other checksum files.

However, I don't think we should do cleanup in this way during the build. In #1231 , we'll remove the directory used to store all of these build-time temporary bits, we don't need to remove them all individually.

This PR will cause merge conflicts with #1231 anyway, so can you revert these changes and I'll handle it in that PR?

Reply:
I see... How about `sudo rm ./*.sha256 ./*.sha512` then? will it still be a conflict?